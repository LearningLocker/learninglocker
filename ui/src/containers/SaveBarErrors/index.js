import React from 'react';
import { compose, withHandlers } from 'recompose';
import { connect } from 'react-redux';
import { getAlertsSelector, deleteAlert } from 'ui/redux/modules/alerts';
import styled from 'styled-components';

const SaveBarErrorsContainer = styled.div`
  top: 2.4rem;
  position: absolute;
  right: 30px;
  left: 25px;
`;

const onRemove = ({ deleteAlert: deleteAlert2 }) => key => () => deleteAlert2({ key });

const SaveBarErrors = ({
  onRemove: onRemove2,
  alerts,
}) => (
  <SaveBarErrorsContainer>
    {
      alerts
        .map(
          (message, key) =>
            <div className="alert alert-danger alert-dismissible" role="alert" key={key}>
              <button onClick={onRemove2(key)} type="button" className="close" data-dismiss="alert" aria-label="Close">
                <span aria-hidden="true">&times;</span>
              </button>
              {message.message}
            </div>
        )
    }
  </SaveBarErrorsContainer>
);

const SaveBarErrorsComposed = compose(
  connect(
    state => ({
      alerts: getAlertsSelector(state)
    }),
    {
      deleteAlert
    }
  ),
  withHandlers({
    onRemove
  })
)(SaveBarErrors);

export default SaveBarErrorsComposed;
