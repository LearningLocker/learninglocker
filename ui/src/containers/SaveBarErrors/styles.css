
.container {
  top: 2.4rem;
  position: absolute;
  right: 30px;
  left: 25px;
}