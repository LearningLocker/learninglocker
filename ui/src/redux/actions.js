export {
  decodeLoginTokenAction
} from 'ui/redux/modules/auth';
export {
  addModel,
  saveModel,
  deleteModel,
  expandModel,
  modelsWorker,
  fetchModel,
  updateModel,
  pollWhile
} from 'ui/redux/modules/models';
export {
  clearModelsCache,
  fetchModels,
  fetchModelsCount,
  fetchMore,
  fetchAllOutstandingModels
} from 'ui/redux/modules/pagination';
export {
  updateStatementQuery,
  updateStatementTimezone
} from 'ui/redux/modules/statements';
export {
  downloadExport
} from 'ui/redux/modules/exports';
export {
  deleteUserOrganisation,
} from 'ui/redux/modules/userOrganisations';
export {
  updateUserOrganisationSetting,
} from 'ui/redux/modules/userOrganisationSettings';
