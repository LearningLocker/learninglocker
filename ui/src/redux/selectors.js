export {
  modelsSelector,
  modelsSchemaSelector,
  modelsSchemaIdSelector,
  modelsPickSelector,
  modelsByFilterSelector,
  isLoadingModelSelector,
  shouldFetchModelSelector,
} from 'ui/redux/modules/models';
export {
  idsByFilterSelector,
  isLoadingSelector,
  isLoadingCountSelector,
  countSelector,
  hasMoreSelector,
  shouldFetchSelector,
  shouldFetchCountSelector,
  cursorSelector
} from 'ui/redux/modules/pagination';
export {
  statementQuerySelector,
  statementTimezoneSelector
} from 'ui/redux/modules/statements';
export {
  isAuthenticatedSelector,
  isLoggingOutSelector,
  isAuthenticatedWithOrgSelector,
  loggedInUserId,
  loggedInUserSelector,
  currentScopesSelector,
  organisationSettingsSelector,
  activeOrganisationSettingsSelector,
  isSiteAdminSelector
} from 'ui/redux/modules/auth';
