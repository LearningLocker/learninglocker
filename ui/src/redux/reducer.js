import { combineReducers } from 'redux';
import recycleState from 'redux-recycle';
import { router5Reducer } from 'redux-router5';

import auth, { LOGOUT, ORG_LOGOUT } from 'ui/redux/modules/auth';
import pagination from 'ui/redux/modules/pagination';
import models from 'ui/redux/modules/models';
import aggregation from 'ui/redux/modules/aggregation';
import statements from 'ui/redux/modules/statements';
import search from 'ui/redux/modules/search';
import uploadpeople from 'ui/redux/modules/people';
import exportsReducer from 'ui/redux/modules/exports';
import logo from 'ui/redux/modules/logo';
import clientIds from 'ui/redux/modules/clientIds';
import toasts from 'ui/redux/modules/toasts';
import queries from 'ui/redux/modules/queries';
import metadata from 'ui/redux/modules/metadata';
import app from 'ui/redux/modules/app';
import alerts from 'ui/redux/modules/alerts';

export default combineReducers({
  auth: recycleState(auth, [LOGOUT]),
  pagination: recycleState(pagination, [LOGOUT, ORG_LOGOUT]),
  aggregation: recycleState(aggregation, [LOGOUT, ORG_LOGOUT]),
  models: recycleState(models, [LOGOUT, ORG_LOGOUT]),
  uploadpeople: recycleState(uploadpeople, [LOGOUT, ORG_LOGOUT]),
  exports: recycleState(exportsReducer, [LOGOUT, ORG_LOGOUT]),
  logo: recycleState(logo, [LOGOUT, ORG_LOGOUT]),
  statements: recycleState(statements, [LOGOUT, ORG_LOGOUT]),
  search: recycleState(search, [LOGOUT, ORG_LOGOUT]),
  toasts: recycleState(toasts, [LOGOUT, ORG_LOGOUT]),
  clientIds: recycleState(clientIds, [LOGOUT, ORG_LOGOUT]),
  queries: recycleState(queries, [LOGOUT, ORG_LOGOUT]),
  metadata: recycleState(metadata, [LOGOUT, ORG_LOGOUT]),
  alerts: recycleState(alerts, [LOGOUT, ORG_LOGOUT]),
  app,
  router: router5Reducer
});
