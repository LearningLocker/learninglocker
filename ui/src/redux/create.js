import { createStore as _createStore, applyMiddleware, compose } from 'redux';
import { router5Middleware } from 'redux-router5';
import clientMiddleware from 'ui/redux/middleware/clientMiddleware';
import thunkMiddleware from 'redux-thunk';
import userMiddleware from 'ui/redux/middleware/userMiddleware';
import rootSaga, { sagaMiddleware } from 'ui/redux/modules/sagas';
import reducer from 'ui/redux/reducer';


export default function createStore(llClient, router, data) {
  const middleware = [
    thunkMiddleware,
    userMiddleware,
    clientMiddleware(llClient),
    sagaMiddleware,
    router5Middleware(router),
  ];

  const composeEnhancers =
    (typeof window === 'object' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__)
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    : compose;

  let finalCreateStore;
  if (__DEVELOPMENT__ && __CLIENT__ && __DEVTOOLS__) {
    const { persistState } = require('redux-devtools');

    finalCreateStore = composeEnhancers(
      applyMiddleware(...middleware),
      persistState(window.location.href.match(/[?&]debug_session=([^&]+)\b/))
    )(_createStore);
  } else {
    finalCreateStore = applyMiddleware(...middleware)(_createStore);
  }

  const store = finalCreateStore(reducer, data);
  sagaMiddleware.run(rootSaga);

  return store;
}

// UNICORN
