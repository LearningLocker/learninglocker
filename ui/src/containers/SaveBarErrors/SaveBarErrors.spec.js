import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { List } from 'immutable';
import SaveBarErrors from './index';

const createMockStore = () => {
  const mockState = {
    alerts: new List([{
      message: 'Hello World'
    }])
  };

  return createStore(
    (data, action) => {
      switch (action.type) {
        case '@@redux/INIT':
          return data;
        default:
          return;
      }
    },
    mockState
  );
};

describe('SaveBarErrors', () => {
  it('should render an error message', () => {
    const mockStore = createMockStore();

    const errors = renderer
      .create(
        <Provider store={mockStore}>
          <SaveBarErrors />
        </Provider>
      )
      .toJSON();

    expect(errors).toMatchSnapshot();
  });
});
