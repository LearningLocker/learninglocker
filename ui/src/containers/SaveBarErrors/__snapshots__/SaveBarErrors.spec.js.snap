// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaveBarErrors should render an error message 1`] = `
.c0 {
  top: 2.4rem;
  position: absolute;
  right: 30px;
  left: 25px;
}

<div
  className="c0"
>
  <div
    className="alert alert-danger alert-dismissible"
    role="alert"
  >
    <button
      aria-label="Close"
      className="close"
      data-dismiss="alert"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        Ã—
      </span>
    </button>
    Hello World
  </div>
</div>
`;
